website/archiver/utils.py

```diff
@@ -111,8 +111,9 @@ def handle_archive_fail(reason, src, dst, user, result):
         pass
     else:  # reason == ARCHIVER_UNCAUGHT_ERROR
         send_archiver_uncaught_error_mails(src, user, result, url)
-    dst.root.sanction.forcibly_reject()
-    dst.root.sanction.save()
+    if dst.root.sanction:
+        dst.root.sanction.forcibly_reject()
+        dst.root.sanction.save()
```

Comment:
Looking at the [`RegistrationFactory`](https://github.com/CenterForOpenScience/osf.io/blob/d07206d3a7b4a26e440c66131531627170e9b1c9/osf_tests/factories.py#L441-L452) and [`ArchiverTestCase`](https://github.com/CenterForOpenScience/osf.io/blob/d07206d3a7b4a26e440c66131531627170e9b1c9/osf_tests/test_archiver.py#L374), `sanction` object passed into shouldn't be `None`.

Here are the failures that probably led to this fix.

```
__________________ TestArchiverUtils.test_handle_archive_fail __________________
osf_tests/test_archiver.py:758: in test_handle_archive_fail
    {}
website/archiver/utils.py:114: in handle_archive_fail
    dst.root.sanction.forcibly_reject()
E   AttributeError: 'NoneType' object has no attribute 'forcibly_reject'
_______________ TestArchiverUtils.test_handle_archive_fail_copy ________________
osf_tests/test_archiver.py:771: in test_handle_archive_fail_copy
    {}
website/archiver/utils.py:114: in handle_archive_fail
    dst.root.sanction.forcibly_reject()
E   AttributeError: 'NoneType' object has no attribute 'forcibly_reject'
_______________ TestArchiverUtils.test_handle_archive_fail_size ________________
osf_tests/test_archiver.py:803: in test_handle_archive_fail_size
    {}
website/archiver/utils.py:114: in handle_archive_fail
    dst.root.sanction.forcibly_reject()
E   AttributeError: 'NoneType' object has no attribute 'forcibly_reject'
```

However, I think the correct fix should be on the RegistrationFactory instead of the actual code. Otherwise, the two lines inside the `if` condition is never tested.